docs: add cert import (#3523)
aws/copilot-cli: site/content/blogs/release-v118.en.md

See the section [Overview](../docs/concepts/overview.en.md) for a more detailed introduction to AWS Copilot.

## Certificate Import

_Contributed by [Penghao He](https://github.com/iamhopaul123/)_

> If you have domains managed outside of Route 53, or want to enable HTTPS without having domain associated with your application, you can now use the new `--import-cert-arns` flag to import any validated certificates when creating your environments.

Reviewer: ```suggestion
If you have domains managed outside of Route 53, or want to enable HTTPS without having a domain associated with your application, you can now use the new `--import-cert-arns` flag to import any validated certificates when creating your environments.
```

```
$ copilot env init --import-cert-arns arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012 --import-cert-arns arn:aws:acm:us-east-1:123456789012:certificate/87654321-4321-4321-4321-210987654321
```

For example, one of the certificates has `example.com` as its domain and `*.example.com` as a subject alternative name (SAN):

```
$ aws acm describe-certificate --certificate-arn arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012

{
  "Certificate": {
    "CertificateArn": "arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012",
    "DomainName": "example.com",
    "SubjectAlternativeNames": [
      "*.example.com"
    ],
    "DomainValidationOptions": [
      {
        "DomainName": "example.com",
        "ValidationDomain": "example.com",
        "ValidationStatus": "SUCCESS",
        "ResourceRecord": {
          "Name": "_45c8aa9ac85568e905a6c3852e62ebc6.example.com.",
          "Type": "CNAME",
          "Value": "_f8be688050b7d23184863690b3d4baa8.xrchbtpdjs.acm-validations.aws."
        },
        "ValidationMethod": "DNS"
      }
    ],
    ...
}
```

> Then, you can specify aliases that are valid against any of the imported certificates in a [Load Balanced Web Service manifest](../docs/manifest/lb-web-service.en.md):

Reviewer 2: Should we mention that this is not optional? Like alias is required for the whole thing to work.

Author: Good call!

```yaml
name: frontend
type: Load Balanced Web Service

http:
  path: 'frontend'
  alias: v1.example.com
```

> After the deployment, the last step to make the alias work is to add the DNS of the Application Load Balancer (ALB) created in the environment as an A record to where your alias domain is hosted. For example, if your alias domain is hosted in Route 53:

Reviewer 2: I felt like this part "to make the alias work" part was a little expected to me because I was reading the earlier parts without assuming "using alias was the ultimate purpose". What do you think of:

```suggestion
After the deployment, add the DNS of the Application Load Balancer (ALB) created in the environment as an A record to where your alias domain is hosted. For example, if your alias domain is hosted in Route 53:
```

 I'm not sure if this is a legit nit. Feel free to disagree!!

```json
{
  "Name": "example.com.",
  "Type": "A",
  "AliasTarget": {
    "HostedZoneId": "Z1H1FL3HABSF5",
    "DNSName": "demo-publi-1d328e3bqag4r-1914228528.us-west-2.elb.amazonaws.com.",
    "EvaluateTargetHealth": true
  }
}
```

> 

Reviewer 2: ```suggestion
Now, your service has HTTPS enabled using your own certificates and can be accessed via "https://example.com"!
```

        "ValidationStatus": "SUCCESS",
        "ResourceRecord": {
          "Name": "_45c8aa9ac85568e905a6c3852e62ebc6.example.com.",
          "Type": "CNAME",
          "Value": "_f8be688050b7d23184863690b3d4baa8.xrchbtpdjs.acm-validations.aws."
        },
        "ValidationMethod": "DNS"
      }
    ],
    ...
}
```

Then, you need to specify aliases that are valid against any of the imported certificates in a [Load Balanced Web Service manifest](../docs/manifest/lb-web-service.en.md):

```yaml
name: frontend
type: Load Balanced Web Service

http:
  path: 'frontend'
  alias: v1.example.com
```

!!!attention
    Specifying `http.alias` in service manifests is required for deploying services to an environment with imported certificates.

After the deployment, add the DNS of the Application Load Balancer (ALB) created in the environment as an A record to where your alias domain is hosted. For example, if your alias domain is hosted in Route 53:

```json
{
  "Name": "example.com.",
  "Type": "A",
  "AliasTarget": {
    "HostedZoneId": "Z1H1FL3HABSF5",
    "DNSName": "demo-publi-1d328e3bqag4r-1914228528.us-west-2.elb.amazonaws.com.",
    "EvaluateTargetHealth": true
  }
}
```

> Now, your service has HTTPS enabled using your own certificates and can be accessed via https://example.com!

Reviewer 3: Should this be 
```suggestion
Now, your service has HTTPS enabled using your own certificates and can be accessed via https://v1.example.com!
```
To match the manifest? similar with the alias record block above

Author: 🤦

See the section [Overview](../docs/concepts/overview.en.md) for a more detailed introduction to AWS Copilot.

## Certificate import

_Contributed by [Penghao He](https://github.com/iamhopaul123/)_

> Developers that have domains managed outside of Route 53 or who want to use HTTPS for Copilot applications without any domain specified can now use a new `--import-cert-arns` flag to import any validated certificates when they create an environment.

Reviewer 2: nit: maybe this will break down the sentence a little bit? 

```suggestion
If you have domains managed outside of Route 53, or want to enable HTTPS without having domain associated with your application, you can now use the new `--import-cert-arns` flag to import any validated certificates when creating your environments.
```

> Developers that have domains not managed by Route 53 or want to use HTTPS for Copilot applications without any domain specified can now use a new `--import-cert-arns` flag to import any validated certificates when they create an environment, so as to use any domain/SAN specified in their imported certificates for their services deployed to the environment:

Reviewer: ```suggestion
Developers that have domains managed outside of Route 53 or who want to use HTTPS for Copilot applications without any domain specified can now use a new `--import-cert-arns` flag to import validated certificates when they create an environment. so as to use any domain/SAN specified in their imported certificates for their services deployed to the environment:
```

i'm not quite understanding the sentence/section starting with "so as to use" - is it supposed to be introducing the example below?

Author: It is supposed to explain how specifying the flag helps them deal with the issue (that they cannot use domain outside of route 53). But I'll remove the sentence instead.

```
$ copilot env init --import-cert-arns arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012
```

For example, the certificate has `example.com` as its domain and `*.example.com` as a subject alternative name (SAN):

```
$ aws acm describe-certificate --certificate-arn arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012

{
  "Certificate": {
    "CertificateArn": "arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012",
    "DomainName": "example.com",
    "SubjectAlternativeNames": [
      "*.example.com"
    ],
    "DomainValidationOptions": [
      {
        "DomainName": "example.com",
        "ValidationDomain": "example.com",
        "ValidationStatus": "SUCCESS",
        "ResourceRecord": {
          "Name": "_45c8aa9ac85568e905a6c3852e62ebc6.example.com.",
          "Type": "CNAME",
          "Value": "_f8be688050b7d23184863690b3d4baa8.xrchbtpdjs.acm-validations.aws."
        },
        "ValidationMethod": "DNS"
      }

>     ],

Reviewer 2: Is this chunk needed for demonstration purpose? Or can it be shortened to reduce noise 🤔 ?

Author: I am still more inclined to keep it, because a blog post I think we should have more concrete examples showing them what they should expect for each step.

Reviewer 3: You can also use https://squidfunk.github.io/mkdocs-material/reference/admonitions/?h=colla#collapsible-blocks so that you can toggle to expand or collapse the code block:
```
???+ example "Sample certificate"

    ```json
    
    ```
```

    ...
}
```

> Then, aliases that are valid against any of the imported certificates can be specified in a [Load Balanced Web Service manifest](../manifest/lb-web-service.en.md) to deploy a service to the environment:

Reviewer 2: ```suggestion
Then, you can specify aliases that are valid against any of the imported certificates in a [Load Balanced Web Service's manifest](../manifest/lb-web-service.en.md):
```

Also, since this sentence mentions `"any of the imported certificates"`, do you think it'd flow better if the example above import two certificates?

> Then, aliases that are valid against any of the imported certificates can be specified in the Load Balanced Web Service manifest to deploy the services to the environment:

Reviewer: Load Balanced Web Service manifest 
```suggestion
Then, aliases that are valid against any of the imported certificates can be specified in a [Load Balanced Web Service manifest](link goes here) to deploy a service to the environment:
```
